### PR TITLE
Create ua module and move ua classes

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205489571133516/f 

### Description
This PR moves the ua provider and plugin to its own module

### Steps to test this PR

- [x] With internal flavour test that all the UA are correctly set.
- [x] Filter the logcat by `user agent is`
- [x] Go to `digid.nl`, UA should not have the duckduckgo component
- [x] Go to `marvel.com` UA should not have the duckduckgo component or version
- [x] Go to `whitepages.com` UA should not have the version component
- [x] Go to `reddit.com` UA should be the WebView default
- [x] Go to `duckduckgo.com` UA should be our custom one

### CODE REVIEW

**Comment by marcosholgado** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3542** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3542" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3542?utm_source=stack-comment).

File: user-agent/user-agent-impl/build.gradle [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
```
Comment (joshliebe) [inline user-agent/user-agent-impl/build.gradle:2]:
```suggestion
 * Copyright (c) 2023 DuckDuckGo
```

File: user-agent/user-agent-api/build.gradle [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
```
Comment (joshliebe) [inline user-agent/user-agent-api/build.gradle:2]:
```suggestion
 * Copyright (c) 2023 DuckDuckGo
```

**Review by joshliebe — APPROVED** [review]
lgtm and tested 👍

**Pull request merged**